tests/test_handlers.py

```diff
@@ -0,0 +1,332 @@
+
+        sch = nonbonded.SimpleChargeHandler(smirks, params)
+
+        mol = Chem.MolFromSmiles("C1CNCOC1F")
+
+        NL = mol.GetNumAtoms()
+        NP = 13
+        aux_es_params = np.random.rand(NP,) + 10
+
+        es_params, es_vjp_fn = sch.parameterize(mol, aux_es_params)
+
+        assert es_params.shape == (NP + NL,)
+
+        np.testing.assert_almost_equal(es_params[NL:], aux_es_params)
+
+        ligand_params = np.array([
+            0.1, # C
+            0.1, # C
+            0.3, # N
+            0.1, # C
+            0.5, # O
+            0.1, # C
+            1.0  # F
+        ])
+
+        np.testing.assert_almost_equal(es_params[:NL], ligand_params)
+
+        es_params_adjoints = np.random.randn(*es_params.shape)
+
+        # test that we can use the adjoints
+        adjoints = es_vjp_fn(es_params_adjoints)[0]
+
+        # if a parameter is > 99 then its adjoint should be zero (converse isn't necessarily true since)
+        mask = np.argwhere(params > 90)
+        assert np.all(adjoints[mask] == 0.0) == True
+
+    def test_lennard_jones_handler(self):
+
+        patterns = [
+            ['[#1:1]', 99., 999.],
+            ['[#1:1]-[#6X4]', 99., 999.],
+            ['[#1:1]-[#6X4]-[#7,#8,#9,#16,#17,#35]', 99., 999.],
+            ['[#1:1]-[#6X4](-[#7,#8,#9,#16,#17,#35])-[#7,#8,#9,#16,#17,#35]', 99., 999.],
+            ['[#1:1]-[#6X4](-[#7,#8,#9,#16,#17,#35])(-[#7,#8,#9,#16,#17,#35])-[#7,#8,#9,#16,#17,#35]', 99., 999.],
+            ['[#1:1]-[#6X4]~[*+1,*+2]', 99., 999.],
+            ['[#1:1]-[#6X3]', 99., 999.],
+            ['[#1:1]-[#6X3]~[#7,#8,#9,#16,#17,#35]', 99., 999.],
+            ['[#1:1]-[#6X3](~[#7,#8,#9,#16,#17,#35])~[#7,#8,#9,#16,#17,#35]', 99., 999.],
+            ['[#1:1]-[#6X2]', 99., 999.],
+            ['[#1:1]-[#7]', 99., 999.],
+            ['[#1:1]-[#8]', 99., 999.],
+            ['[#1:1]-[#16]', 99., 999.],
+            ['[#6:1]', 0.7, 0.8],
+            ['[#6X2:1]', 99., 999.],
+            ['[#6X4:1]', 0.1, 0.2],
+            ['[#8:1]', 99., 999.],
+            ['[#8X2H0+0:1]', 0.5, 0.6],
+            ['[#8X2H1+0:1]', 99., 999.],
+            ['[#7:1]', 0.3, 0.4],
+            ['[#16:1]', 99., 999.],
+            ['[#15:1]', 99., 999.],
+            ['[#9:1]', 1.0, 1.1],
+            ['[#17:1]', 99., 999.],
+            ['[#35:1]', 99., 999.],
+            ['[#53:1]', 99., 999.],
+            ['[#3+1:1]', 99., 999.],
+            ['[#11+1:1]', 99., 999.],
+            ['[#19+1:1]', 99., 999.],
+            ['[#37+1:1]', 99., 999.],
+            ['[#55+1:1]', 99., 999.],
+            ['[#9X0-1:1]', 99., 999.],
+            ['[#17X0-1:1]', 99., 999.],
+            ['[#35X0-1:1]', 99., 999.],
+            ['[#53X0-1:1]', 99., 999.],
+        ]
+
+        smirks = [x[0] for x in patterns]
+        params = np.array([[x[1], x[2]] for x in patterns])
+
+        ljh = nonbonded.LennardJonesHandler(smirks, params)
+
+        mol = Chem.MolFromSmiles("C1CNCOC1F")
```

Comment:
Should this (and other similar mols in tests) have hydrogens?

Reply:
Apparently for testing purposes it doesn't matter. It just adds the number of atoms I need to manually check. But you're right that in production we need to do something like AddHs